.werft/util/werft.ts

```diff
@@ -20,9 +20,11 @@ export class Werft {
     private tracer: Tracer;
     public rootSpan: Span;
     private sliceSpans: { [slice: string]: Span } = {}
-    public currentPhaseSpan: Span;
+    public currentPhase: string;
     private globalSpanAttributes: SpanAttributes = {}
 
+    public phases: { [name: string]: { "span": Span, "closed": boolean, "slices": { [name: string]: { "closed": boolean, "span": Span } } } } = {}
```

Comment:
Can you introduce a Type for this datastructure?

Reply:
I can do that. I already proposed that in the issue [[1](https://github.com/gitpod-io/gitpod/issues/8707#issuecomment-1085573593)].

Reply:
I created a light version of my datastructures for phases and slices to solve this.